### PR TITLE
Update SwiftLint to 0.53.0 and SwiftFormat to 0.52.4

### PR DESCRIPTION
**Pull request by peagasilva**
Child PR: https://github.com/WeTransfer/Mule/pull/3289.

This PR updates both SwiftLint and SwiftFormat to their latest versions.

### Related Jira Issue
This PR fixes [TMOB-1798].

[TMOB-1798]: https://wetransfer.atlassian.net/browse/TMOB-1798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BuildTools/.swiftlint.yml [hunk 1]
```diff
@@ -90,6 +88,10 @@ type_contents_order:
     - [subscript, other_method, ib_action]
     - type_method
 
+analyzer_rules:
+  - capture_variable
+  - unused_import
```
Comment (kairadiagne) [inline BuildTools/.swiftlint.yml:93]: Nice!

**Review by kairadiagne — APPROVED** [review]

File: BuildTools/.swiftformat [hunk 2]
```diff
@@ -1,8 +1,8 @@
 --indent 4
 --patternlet inline
---disable unusedArguments
+--disable unusedArguments,hoistAwait,hoistTry
```
Comment (BasThomas) [inline BuildTools/.swiftformat:3]: Why did we decide to disable these?

Comment (peagasilva) [reply BuildTools/.swiftformat:3]: They are not quite there yet:

image

File: WeTransferPRLinter/Sources/WeTransferPRLinter/XCResultReporting/Extensions/ErrorExtensions.swift [hunk 3]
```diff
@@ -5,9 +5,9 @@ extension LocationError {
     var isMissingError: Bool {
         switch reason {
         case .missing:
-            return true
+            true
         default:
-            return false
+            false
```
Comment (AvdLee) [inline WeTransferPRLinter/Sources/WeTransferPRLinter/XCResultReporting/Extensions/ErrorExtensions.swift:10]: Nit: you could write this as follows as well:

```swift
case .missing: true
default: false
```

**Review by AvdLee — APPROVED** [review]
This looks good to me! I do think we need to create a new binary of our Danger binary as well ([as explained in the readme](https://github.com/WeTransfer/WeTransfer-iOS-CI/commit/a04da310baeab00680f844b08e2a6e84793e21fa)) since SwiftLint is packaged within Danger IIRC.

**Comment by peagasilva** [comment]
> This looks good to me! I do think we need to create a new binary of our Danger binary as well ([as explained in the readme](https://github.com/WeTransfer/WeTransfer-iOS-CI/commit/a04da310baeab00680f844b08e2a6e84793e21fa)) since SwiftLint is packaged within Danger IIRC.

Thanks for the heads up, @AvdLee! I've just updated the binary file and the README file.

**Pull request merged**